### PR TITLE
Hide the "show guests only" checkbox in user modules

### LINKED ISSUE #823: Feld "Nur Gästen anzeigen" bei Frontend-Modulen für Mitglieder
Beim Bearbeiten von Frontend-Modulen für Mitglieder (z. B. "Personendaten" oder "Passwort ändern") wird das Feld "Nur Gästen anzeigen" in der Rubrik Experteneinstellungen angezeigt, obwohl diese Module ausschließlich angemeldeten Mitgliedern angezeigt werden.

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #823

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

**Review by aschempp — CHANGES_REQUESTED** [review]
I think we should not remove the `protected` option for login module. If the login module can only be shown for guest, one might need an matching login module only for users and/or a different one (template) for different groups.

**Comment by leofeyer** [comment]
What exactly do you mean?

**Comment by Mynyx** [comment]
I do not understand his usecase, but in opposite to the other modules the login module is shown to authenticated users too. Therefore the ``protected`` field should not be removed.

File: core-bundle/src/Resources/contao/dca/tl_module.php [hunk 1]
```diff
@@ -106,13 +106,13 @@
...
-		'changePassword'              => '{title_legend},name,headline,type;{redirect_legend},jumpTo;{template_legend:hide},customTpl;{protected_legend:hide},protected;{expert_legend:hide},guests,cssID',
+		'login'                       => '{title_legend},name,headline,type;{config_legend},autologin;{redirect_legend},jumpTo,redirectBack;{template_legend:hide},cols,customTpl;{expert_legend:hide},guests,cssID',
+		'logout'                      => '{title_legend},name,type;{redirect_legend},jumpTo,redirectBack;{protected_legend:hide},protected;{expert_legend:hide},cssID',
+		'personalData'                => '{title_legend},name,headline,type;{config_legend},editable;{redirect_legend},jumpTo;{template_legend:hide},memberTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
+		'registration'                => '{title_legend},name,headline,type;{config_legend},editable,newsletters,disableCaptcha;{account_legend},reg_groups,reg_allowLogin,reg_assignDir;{redirect_legend},jumpTo;{email_legend:hide},reg_activate;{template_legend:hide},memberTpl;{expert_legend:hide},guests,cssID',
```
Comment (Mynyx) [inline core-bundle/src/Resources/contao/dca/tl_module.php]:
```suggestion
		'registration'                => '{title_legend},name,headline,type;{config_legend},editable,newsletters,disableCaptcha;{account_legend},reg_groups,reg_allowLogin,reg_assignDir;{redirect_legend},jumpTo;{email_legend:hide},reg_activate;{template_legend:hide},memberTpl;{expert_legend:hide},cssID',
```

**Comment by leofeyer** [comment]
> the login module is shown to authenticated users too

Yes, but it is shown to **all** users who can log in. It is not shown to everyone before login and then no longer to a certain user group after login. What would be the use case?

**Comment by aschempp** [comment]
> > the login module is shown to authenticated users too
> 
> Yes, but it is shown to **all** users who can log in. It is not shown to everyone before login and then no longer to a certain user group after login. What would be the use case?

Not exactly sure what you mean. The login module shows the current logged in user and a logout option. It works for both logged in and logged out users.

**Comment by leofeyer** [comment]
Exactly. That is why it does not make sense to have a "protected" option for the login module. But you seem to request exactly this:

> I think we should not remove the `protected` option for login module.

**Comment by aschempp** [comment]
> Exactly. That is why it does not make sense to have a "protected" option for the login module. But you seem to request exactly this:
> 
> > I think we should not remove the `protected` option for login module.

Yes I do. I have a login module for guests to log in (e.g. in the header). Then I have another login/logout module for members, e.g. in the right column. I need the `protected` option to hide the module in the right column from guests.

**Comment by leofeyer** [comment]
How does this work?

* First I am a guest and I log in in the header log in. 
* Now I am an authenticated user.
* Now there is another login module in the right column – which username do I use?
* And after I have logged in again, am I now logged in as two different users at the same time?

**Comment by aschempp** [comment]
Please read more carefully. The login module shows the username and a logout button if a user is currently logged in 😉

**Comment by leofeyer** [comment]
Ah, I see what you are doing.

I think abusing the login module like this is as wrong as it can be, but at least it is a use case. 😄 A logout page and a (custom) navigation module would be more appropriate IMHO.

File: core-bundle/src/Resources/contao/dca/tl_module.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Resources/contao/dca/tl_module.php]: @Mynyx Why should the guests checkbox be removed here? IMO it maks sense to show a registration module only to guests.

**Review by ausi — DISMISSED** [review]
Approved except for the removal of `protected` for the login module.

File: core-bundle/src/Resources/contao/dca/tl_module.php (hunk 1 above)
Comment (Mynyx) [reply core-bundle/src/Resources/contao/dca/tl_module.php]: I thought the module is only shown to guests and the guests checkbox is redundant. But I have just noticied that this isn't correct.

On the other hand this means, that the field ``protected`` should not be removed as proposed in the pull request.

Comment (aschempp) [reply core-bundle/src/Resources/contao/dca/tl_module.php]: as far as I can tell, the registration module actually _is_ only shown to guests? Or how's that not the case?

**Comment by leofeyer** [comment]
I have re-added the "protected" field to the login module in 5031b1de7b0fe6b8cfbf61ba96d86c577c2ec5d0.

**Comment by leofeyer** [comment]
What about @Mynyx's [comment above](https://github.com/contao/contao/pull/883#pullrequestreview-308046554)? Do we need the "show guests only" option for the registration module? And if we do, do we also need the "protected" option?

**Comment by aschempp** [comment]
Apparently the module is visible to all users, couldn't find anything in https://github.com/contao/contao/blob/master/core-bundle/src/Resources/contao/modules/ModuleRegistration.php
Looks very strange, but apparently we need both checkboxes…

**Comment by leofeyer** [comment]
Re-added in 82a42ae8ca60d69e1e45f1b37cad61e8427996fa.

**Review by ausi — APPROVED** [review]

**Pull request merged**